ssh-krb-node-executor-plugin/contents/scp-krb-executor.sh

```diff
@@ -13,36 +13,49 @@ shift
 SRC_FILE=$1
 shift
 DEST_FILE=$1
-KERB_KEYTAB=$RD_CONFIG_KERBEROS_KEYTAB
 
-if [ ! -f "$KERB_KEYTAB" ]; then
-  echo "Keytab $KERB_KEYTAB not found" >&2
+KRB_USER=$RD_CONFIG_KERBEROS_USER
+KRB_KEYTAB=$RD_CONFIG_KERBEROS_KEYTAB
+USE_CUSTOM_KRB_CACHE=$RD_CONFIG_USE_KERBEROS_CUSTOM_CACHE_FILE
+KRB_CACHE_FILENAME=$RD_CONFIG_KERBEROS_CUSTOM_CACHE_FILENAME
+KRB_DESTROY=$RD_CONFIG_DO_KDESTROY
+
+
+if [ ! -f "$KRB_KEYTAB" ]; then
+  echo "Keytab $KRB_KEYTAB not found" >&2
   exit 2
 fi
 
+
 # random delay (0..0.5s) to make it work with parallel exec
 sleep "$(bc <<< "scale=2; $(printf '0.%02d' $(( RANDOM % 100))) / 2")"
 
-# status 1 if the credentials cache cannot be read or is expired, and with status 0 otherwise
-klist -s
-if [ $? -ne 0 ]; then
-  kinit -kt "$KERB_KEYTAB" "$RD_CONFIG_KERBEROS_USER"
 
-  # verify ticket has been successfuly created
-  klist -s
-  if [ $? -ne 0 ]; then
-    echo "Kinit failure when calling kinit -kt $KERB_KEYTAB $RD_CONFIG_KERBEROS_USER" >&2
+# override the default kerberos cache file if required
+if [[ "$USE_CUSTOM_KRB_CACHE" == 'true' ]]; then
+  if [ ! -z "$KRB_CACHE_FILENAME" ]; then
+    export KRB5CCNAME="$KRB_CACHE_FILENAME"
+  else
+    export KRB5CCNAME="/tmp/krb5cc_$(id -ru)_rundeck_$KRB_USER"
+  fi
+fi
+
+
+# recreate the cache if it's expired or not present
+if ! klist -s; then
+  kinit -kt $KRB_KEYTAB $KRB_USER
```

Comment:
```suggestion
  kinit -kt "$KRB_KEYTAB" "$KRB_USER"
```

I think it might be better keep double quoting those variables as for instance the path to the keytab might have spaces.

Reply:
[done](https://github.com/cernops/rundeck-exec-kerberos/pull/7/commits/6dd2dc232cd11d973862b955f50f8b0e0a4d02e6)

I also did it for the filenames in `scp` and used `eval` for dealing with the *multi-layer* escaping of the destination file.